h/security/policy/_cookie.py

```diff
@@ -14,13 +14,46 @@ class CookiePolicy:
     straps the login for the client (when the popup shows).
     """
 
-    def __init__(self, cookie: SignedCookieProfile, helper: AuthTicketCookieHelper):
-        self.cookie = cookie
+    def __init__(
+        self,
+        html_authcookie: SignedCookieProfile,
+        api_authcookie: SignedCookieProfile,
+        helper: AuthTicketCookieHelper,
+    ):
+        self.html_authcookie = html_authcookie
+        self.api_authcookie = api_authcookie
         self.helper = helper
 
     def identity(self, request):
         self.helper.add_vary_by_cookie(request)
-        return self.helper.identity(self.cookie, request)
+        identity = self.helper.identity(self.html_authcookie, request)
+
+        # If a request was successfully authenticated using the HTML auth
+        # cookie and that request did *not* also contain the API auth cookie,
+        # then add the API auth cookie to the user's browser.
+        #
+        # This is a temporary hack that was needed when we first added the
+        # separate API auth cookie: we needed to add the API auth cookie to the
+        # browsers of users who were already logged in with just the HTML auth
+        # cookie, we couldn't just rely on logging in to set the API auth
+        # cookie for users who were *already* logged in.
+        #
+        # This code should be deleted after it has been deployed to production for
+        # at least 30 days (the max_age of the HTML auth cookie).
+        #
+        # When deleting this code we should also change the `path` attribute of
+        # api_authcookie to "/api/" so that the API auth cookie is only sent
+        # with API requests. For already-logged-in users this path change won't
+        # take effect until they delete and re-create their API auth cookie by
+        # logging out and in again, which will take at most 30 days (the
+        # max_age of the cookie).
+        #
+        # When changing the `path` attribute of api_authcookie to "/api/" we
+        # should also remove the api_authcookie from the test requests in
+        # _cookie_test.py (see corresponding comment in _cookie_test.py).
+        self._issue_api_authcookie(identity, request)
```

Comment:
I don't think this would ever be a problem, no. When the users HTML auth cookie expires they may find themselves with an API auth cookie but no HTML one. They'll be unable to do anything with their API auth cookie because if they try to load any h pages they'll find themselves logged out (due to the expired HTML auth cookie). If they log back in they'll be issued new HTML and API auth cookies with matching expiry times.